### ISSUE TITLE
Navigating to step picks the first partially matching step

### ISSUE THREAD

**Issue by mvindahl** [alexkrechik/VSCucumberAutoComplete#316]
**Describe the bug**
When navigating to a step, it chooses to go to the first partially matching definition instead of the fully matching definition. This seems to be dictated by order in which steps were scanned.

**To Reproduce**
Example:
```typescript
Then('I see {string}', (text) => {
  // ...
});

Then('I see {string} to be {string}', (label, value) => {
  // ...
});
```

.feature file
```
    ...
    Then I see "foo" to be "bar"
    ...
```
When executing cypress, it has no problem locating the corresponding step but when I use CMD+click to navigate in VSCode, it goes to the first of the two steps.

If I reverse the order in which they were defined, it will work as expected so I assume that it finds the first matching step but that it's too lax in its matching.

**Expected behavior**
It will navigate to the correct step definition no matter in which order they were defined.

Note that as long as all steps live in the same file, a workaround is to be explicit about the ordering of these. In a real scenario with multiple steps files scattered in a folder structure, the order of loading becomes a bit undefined so reordering will not really work.

**Cucumberautocomplete part of VSCode settings:**
```
    "cucumberautocomplete.steps": [
        "cypress/integration/**/*.ts",
        "node_modules/qa-lib/src/step_definitions/*.js"
    ],
    "cucumberautocomplete.syncfeatures": "cypress/features-wip/*feature",
    "cucumberautocomplete.strictGherkinCompletion": true,
    "cucumberautocomplete.strictGherkinValidation": true,
```

**Step definition:**
(See above)

**Gherkin step line**
(See above)

**Comment by AlexKTest** [comment]
Looks like it's not an issue due to there are no regex end symbol in the step definition.

**Comment by alexkrechik** [comment]
Looks like it's not an issue due to there are no regex end symbol in the step definition.

**Comment by CCninja86** [comment]
This is an issue, because as of Cucumber 3, you can use Cucumber Expressions OR regex. If using Cucumber expressions, there are no regex start/end symbols. It would be nice to have a setting that enables strict matching of step definitions.

**Comment by sadunaresh** [comment]
I'm seeing an issue on the same lines

I have this in my feature file X

```
Feature: X

    Scenario: X End point returns X response
        Given the microservice is running
        When I connect to the '/api/v1/settings' endpoint
        Then it should respond with HTTP status 200
        And it should contains a response having application name as 'httpapi'
```

and this in feature file Y

```
Feature: Y

Scenario: Y point returns Y response
    Given the microservice is running
    When I connect to the '/Error' endpoint
    Then it should respond with HTTP status 500
    And it should respond with eid
```

each have their own step definition files say ``X.cs`` and ``Y.cs``.

when I try to navigate to step definition of  "Given the microservice is running" from feature file Y, it navigates to step definition on "X.cs" as that comes first in the order.

**Comment by drweb86** [comment]
Please fix it, we can't use expressions because of it.

**Comment by drweb86** [comment]
@alexkrechik Alex, can you please take a look.

**Comment by alexkrechik** [comment]
Agree, I'll add some option to perform a strict match of steps for such cases.

**Comment by ortsevlised** [comment]
Hi @alexkrechik do you have any update on this issue?
thanks!

**Comment by alexkrechik** [comment]
Looks like same issue with https://github.com/alexkrechik/VSCucumberAutoComplete/issues/222

**Comment by alexkrechik** [comment]
Should be fixed with `"cucumberautocomplete.pureTextSteps": true` option using after 2.16.0 release.

**Comment by alexkrechik** [comment]
This issue is to be fixed or implemented in the 3.0.0 version of the extension. If this issue is unresolved or there are additional notes, please create a new ticket with the necessary details.
